addons/mail/models/mail_thread.py

```diff
@@ -1434,6 +1439,12 @@ def _message_parse_extract_payload(self, message, save_original=False):
                 else:
                     attachments.append(self._Attachment(filename or 'attachment', part.get_content(), {}))
 
+                if message.get_content_type() == 'multipart/report' and body:
```

Comment:
we can't move it because we need body to be set

Reply:
You can, it will just break at next iteration instead of the end of the current one. By default body is void, then once it is set it is set for further loops, unless I misunderstood something ?